### PR TITLE
Search User API

### PR DESCRIPTION
**Pull request by HuyVu18**
Call API search User by keyword in last-name or first-name and show 20 result in 1 page.

### CODE REVIEW

**Review by thanhnguyen0311 — APPROVED** [review]

**Review by chphungphat — APPROVED** [review]

File: src/main/java/com/cg_vibely_social_service/controller/UserController.java [hunk 1]
```diff
@@ -101,4 +102,15 @@ public ResponseEntity<?> editUserInfo(@Valid @RequestBody
             return new ResponseEntity<>(HttpStatus.OK);
         }
     }
+
+    @GetMapping("/search")
+    public ResponseEntity<?> searchUser(@RequestParam String keyword,
+                                        @RequestParam Integer number) {
```
Comment (ChungNguyenVr912) [inline src/main/java/com/cg_vibely_social_service/controller/UserController.java]: What is this parameter mean:  @RequestParam Integer number    (page?)

File: src/main/java/com/cg_vibely_social_service/repository/UserRepository.java [hunk 2]
```diff
@@ -12,11 +12,14 @@
...
+
     @Query(value = "SELECT DISTINCT u FROM User u " +
             "WHERE u.id != :userId " +
             "AND u.id NOT IN (SELECT f.friendId FROM Friend f WHERE f.userId = :userId) " +
             "AND u.id NOT IN (SELECT f.userId FROM Friend f WHERE f.friendId = :userId) " +
             "ORDER BY RAND()")
     List<User> findFriendSuggestionByUserId(@Param("userId") Long userId, Pageable pageable);
 
+    @Query(value = "SELECT u FROM User u WHERE u.lastName LIKE %:keyword% OR u.firstName LIKE %:keyword%")
+    List<User> findUsersByLastNameOrFirstName(String keyword, Pageable pageable);
```
Comment (ChungNguyenVr912) [inline src/main/java/com/cg_vibely_social_service/repository/UserRepository.java]: With "pageable" parameter, this method should be return  "Page<User>"

**Review by ChungNguyenVr912 — CHANGES_REQUESTED** [review]

File: src/main/java/com/cg_vibely_social_service/controller/UserController.java (hunk 1 above)
Comment (HuyVu18) [reply src/main/java/com/cg_vibely_social_service/controller/UserController.java]: Resolve!

File: src/main/java/com/cg_vibely_social_service/repository/UserRepository.java (hunk 2 above)
Comment (HuyVu18) [reply src/main/java/com/cg_vibely_social_service/repository/UserRepository.java]: Resolve!

**Review by ChungNguyenVr912 — APPROVED** [review]

**Pull request merged**